public/app/features/explore/hooks/useStateSync/index.ts

```diff
@@ -10,7 +10,7 @@ import { MIXED_DATASOURCE_NAME } from 'app/plugins/datasource/mixed/MixedDataSou
 import { addListener, ExploreItemState, ExploreQueryParams, useDispatch, useSelector } from 'app/types';
```

Comment:
you probably want to add 

```ts
              if (update.panelsState && panelsState) {
                dispatch(changePanelsStateAction({ exploreId, panelsState }));
              }
```

to this hook (around line 135) so that when browser history changes cause a some columns to be selected/deselected the UI is updated accordingly

Reply:
i noticed however that even with this, navigating back in history does never change betwe logs and table, you may want to double check this

Reply:
Yeah we weren't updating the component state when the props changed, should be fixed in the latest